data-prepper-plugins/lambda-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/lambda/accumlator/InMemoryBuffer.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.sink.lambda.accumlator;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.lang3.time.StopWatch;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.services.lambda.LambdaClient;
+import software.amazon.awssdk.services.lambda.model.InvokeRequest;
+import software.amazon.awssdk.services.lambda.model.InvokeResponse;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A buffer can hold in memory data and flushing it to S3.
+ */
+public class InMemoryBuffer implements Buffer {
+
+    private static final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+
+    private final LambdaClient lambdaClient;
+    private final String functionName;
+    private final String invocationType;
+    private int eventCount;
+    private final StopWatch watch;
+    private boolean isCodecStarted;
+
+
+    public InMemoryBuffer(LambdaClient lambdaClient, String functionName, String invocationType) {
+        this.lambdaClient = lambdaClient;
+        this.functionName = functionName;
+        this.invocationType = invocationType;
+
+        byteArrayOutputStream.reset();
+        eventCount = 0;
+        watch = new StopWatch();
+        watch.start();
+        isCodecStarted = false;
+    }
+
+    @Override
+    public long getSize() {
+        return byteArrayOutputStream.size();
+    }
+
+    @Override
+    public int getEventCount() {
+        return eventCount;
+    }
+
+    public Duration getDuration() {
+        return Duration.ofMillis(watch.getTime(TimeUnit.MILLISECONDS));
+    }
+
+    /**
+     * Upload accumulated data to s3 bucket.
+     */
+    @Override
+    public void flushToLambda() {
```

Comment:
e2e acknowledgment seems missing in this PR.

Reply:
e2e ack is handled. refer releaseEventHandles in the pr.